src/plugins/Url/index.js

```diff
@@ -0,0 +1,133 @@
+const Plugin = require('../../core/Plugin')
+const { h } = require('preact')
+const Provider = require('../Provider')
+// const Utils = require('../../core/Utils')
+require('whatwg-fetch')
+
+/**
+ * Link
+ *
+ */
+module.exports = class Link extends Plugin {
+  constructor (uppy, opts) {
+    super(uppy, opts)
+    this.id = this.opts.id || 'Url'
+    this.title = 'Link'
+    this.type = 'acquirer'
+    this.icon = () => <svg aria-hidden="true" class="UppyIcon UppyModalTab-icon" width="64" height="64" viewBox="0 0 64 64">
+      <circle cx="32" cy="32" r="31" />
+      <g fill-rule="nonzero" fill="#FFF">
+        <path d="M25.774 47.357a4.077 4.077 0 0 1-5.76 0L16.9 44.24a4.076 4.076 0 0 1 0-5.758l5.12-5.12-1.817-1.818-5.12 5.122a6.651 6.651 0 0 0 0 9.392l3.113 3.116a6.626 6.626 0 0 0 4.699 1.943c1.7 0 3.401-.649 4.697-1.943l10.241-10.243a6.591 6.591 0 0 0 1.947-4.696 6.599 6.599 0 0 0-1.947-4.696l-3.116-3.114-1.817 1.817 3.116 3.114a4.045 4.045 0 0 1 1.194 2.88 4.045 4.045 0 0 1-1.194 2.878L25.774 47.357z" />
+        <path d="M46.216 14.926a6.597 6.597 0 0 0-4.696-1.946h-.001a6.599 6.599 0 0 0-4.696 1.945L26.582 25.167a6.595 6.595 0 0 0-1.947 4.697 6.599 6.599 0 0 0 1.946 4.698l3.114 3.114 1.818-1.816-3.114-3.114a4.05 4.05 0 0 1-1.194-2.882c0-1.086.424-2.108 1.194-2.878L38.64 16.744a4.042 4.042 0 0 1 2.88-1.194c1.089 0 2.11.425 2.88 1.194l3.114 3.114a4.076 4.076 0 0 1 0 5.758l-5.12 5.12 1.818 1.817 5.12-5.122a6.649 6.649 0 0 0 0-9.393l-3.113-3.114-.003.002z" />
+      </g>
+    </svg>
+
+    // set default options
+    const defaultOptions = {}
+
+    this.opts = Object.assign({}, defaultOptions, opts)
+
+    this.hostname = this.opts.host
+
+    // Bind all event handlers for referencability
+    ;['getMeta', 'handleClick'].forEach(method => {
+      this[method] = this[method].bind(this)
+    })
+
+    this[this.id] = new Provider(uppy, {
```

Comment:
yeah, I'm at crossroads with this one. Here are a few thoughts I have:

1. I think we should consider removing [plugins/Provider](https://github.com/transloadit/uppy/tree/master/src/plugins/Provider) from the plugins folder, because it is not a plugin. And it's the only defaulter there. Maybe we should move it to core? Or somewhere else?

2. Even if we do move [plugins/Provider](https://github.com/transloadit/uppy/tree/master/src/plugins/Provider) to a separate folder, I think its use doesn't apply to the `URL` plugin. I don't consider the URL a Provider because, for GoogleDrive, Dropbox, etc, we have a definite source (aka Provider) they connect to (e.g drive.google.com for GoogleDrive). So the expected behavior can be managed for each Provider.

But for URLs, it would be connecting to different sources (depending on the URL provided), from which we can expect different behaviors (e.g some may require authorization headers etc). So it might be tricky to squeeze that into a Provider. Instead maybe we should have URL stand as a separate plugin and not considered it a Provider type.

3. That said, I think we need a separate module that manages communication with uppy-server. So instead of just [plugins/Provider](https://github.com/transloadit/uppy/tree/master/src/plugins/Provider) which does it right now. We can have a folder structure like this:

```
 ServerBridge(or a better name)
      --- Provider
      --- ApiCalls (or whatever)
```

where plugins like google drive can go on to use `Provider`, while plugins like URL, S3 can just use ApiCalls directly. And then going back to suggestion **1.** we could move `ServerBridge` to `Core` or make it standalone.

What do you think? I hope I made sense 😄

Reply:
We discussed this in the call yesterday and decided to try the following: add generic API calls to `provider/index.js` so that we don’t have to repeat the boilerplate with fetch headers and so on. For now leave things as one this, provider, maybe think on a more generic name, like uppy-server-api or something.

Then maybe provider lives in `uppy-core`, but `uppy-provider-views` is a separate package.